microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/TracingDeploymentProcessor.java

```diff
@@ -158,18 +158,40 @@ private String getTracerConfiguration(DeploymentPhaseContext deploymentPhaseCont
         return null;
     }
 
-    private void addContextParameter(JBossWebMetaData jbossWebMetaData, ParamValueMetaData restEasyProvider) {
+    private void addResteasyProvidersParameter(JBossWebMetaData jbossWebMetaData) {
         if (jbossWebMetaData == null) {
             return;
         }
         List<ParamValueMetaData> contextParams = jbossWebMetaData.getContextParams();
         if (null == contextParams) {
             contextParams = new ArrayList<>();
         }
-        contextParams.add(restEasyProvider);
+        ParamValueMetaData restEasyDynamicFeature = getResteasyProvidersParam(jbossWebMetaData);
+        if (restEasyDynamicFeature.getParamValue() != null && !restEasyDynamicFeature.getParamValue().trim().isEmpty()) {
+            restEasyDynamicFeature.setParamValue(restEasyDynamicFeature.getParamValue() + ",org.wildfly.microprofile.opentracing.smallrye.TracerDynamicFeature");
+        } else {
+            restEasyDynamicFeature.setParamValue("org.wildfly.microprofile.opentracing.smallrye.TracerDynamicFeature");
+        }
+        contextParams.add(restEasyDynamicFeature);
```

Comment:
This may add a 2nd 'resteasy.providers' ParamValueMetaData to the list, which may work but seems fragile.

Reply:
correct

Reply:
@ehsavoie So will it stay this way or do you expect to change it?

Reply:
I'm fixing this and updating the test as it wasn't breaking without the fix ;)